src/web/mjs/connectors/Sukima.mjs

```diff
@@ -0,0 +1,191 @@
+                    pages = pageContent.max_page;
+
+                    for (const manga of pageContent.items) {
+                        mangas.push(
+                            {
+                                id: '/api/book/v1/title/'+manga.title_code+'/',
+                                title: manga.title_name
+                            }
+                        );
+                    }
+                }
+            }
+        }
+
+        return mangas;
+    }
+
+    async _getChapters(manga) {
+        let request = new Request(new URL(manga.id, this.url), {
+            method: 'POST',
+            body: '{}',
+            headers: {
+                'Content-Type': 'application/json;charset=utf-8',
+            }
+        });
+
+        let chapters = [];
+        let books = await this.fetchJSON(request);
+        for (const book of books.contents) {
+            for (const chapter of book.stories) {
+                chapters.push(
+                    {
+                        id: '/bv/t/' + chapter.title_code + '/v/' + chapter.volume + '/s/' + chapter.story + '/p/0',
+                        title: chapter.volume.toString().padStart(3, '0') + '-' + chapter.story.toString().padStart(3, '0') + ' - ' + chapter.info.text.trim()
+                    }
+                );
+            }
+        }
+        return chapters;
+    }
+
+    async _getPages(manga) {
+        let script = `
+            new Promise((resolve, reject) => {
+                setTimeout(() => {
+                    let shuffle_data = [];
+                    for (page = 0; page < PAGES_INFO.length; page++) {
+                        shuffle_data.push(
+                            {
+                                id: PAGES_INFO[page].page_url,
+                                blocklen: BLOCKLEN,
+                                shuffle_map: JSON.parse(PAGES_INFO[page].shuffle_map)
+                            }
+                        )
+                    }
+                    resolve(shuffle_data);
+                }, 2500);
+            });
+        `;
+        let request = new Request(new URL(manga.id, this.url));
+        let data = await Engine.Request.fetchUI(request, script);
+
+        return data.map(page => {
+            return this.createConnectorURI(
+                {
+                    id: page.id,
+                    blocklen: page.blocklen,
+                    shuffle_map: page.shuffle_map,
+                }
+            );
+        });
+    }
+
+    async _handleConnectorURI(payload) {
+        while (this.lock) {
+            await this.throttle_download( Math.floor(Math.random() * (2000 - 1000)) + 1000);
+        }
+        this.lock = true;
+
+        let request = new Request(new URL(payload.id), this.requestOptions);
+        let response = await fetch(request);
+        let blob = await response.blob();
+        let bitmap = await createImageBitmap(blob);
+
+        let canvas = document.createElement('canvas');
```

Comment:
May extract the bitmap descrambling code to an async method that returns the descrambled blob

Reply:
Would make sense if I'm aiming for parallel processing, but I'm intentionally wait for the processing to finish on the current image.

Reply:
You can still await when calling the method
```javascript
_handleConnectorURI(payload) {
    let blob = await response.blob();
    blob = await this._descrambleImage(blob, blocks, key, ...);
    return blob;
}

async _descrambleImage(imageData, blocks, key, ...) {
    // ...
}
```